src/test/java/com/redhat/devtools/lsp4ij/client/SettingsHelperTest.java

```diff
@@ -0,0 +1,73 @@
+package com.redhat.devtools.lsp4ij.client;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.intellij.testFramework.fixtures.BasePlatformTestCase;
+
+import java.io.StringReader;
+
+public class SettingsHelperTest extends BasePlatformTestCase {
+    // language=json
+    private final String testJson = """
+            {
+                "mylsp": {
+                    "myscalarsetting": "value",
+                    "myobjectsettings": {
+                        "subsettingA": 1,
+                        "subsettingB": 2
+                    }
+                }
+            }
+            """;
+
+    public void testGetSettingsPrimitiveValue() {
+        JsonObject jsonObject = JsonParser.parseReader(new StringReader(testJson)).getAsJsonObject();
+        String[] requestedPath = new String[]{"mylsp", "myscalarsetting"};
+
+        JsonElement result = SettingsHelper.findSettings(requestedPath, jsonObject);
+
+        assertNotNull(result);
+        assertTrue(result.isJsonPrimitive());
+        assertEquals("value", result.getAsString());
+    }
+
+    public void testGetSettingsObjectValue() {
+        JsonObject jsonObject = JsonParser.parseReader(new StringReader(testJson)).getAsJsonObject();
+        String[] requestedPath = new String[]{"mylsp", "myobjectsettings"};
+
+        JsonElement result = SettingsHelper.findSettings(requestedPath, jsonObject);
+
+        assertNotNull(result);
+        assertTrue(result.isJsonObject());
+        assertEquals(1, result.getAsJsonObject().get("subsettingA").getAsInt());
+        assertEquals(2, result.getAsJsonObject().get("subsettingB").getAsInt());
+    }
+
+    public void testGetSettingsNonExistingValue() {
+        JsonObject jsonObject = JsonParser.parseReader(new StringReader(testJson)).getAsJsonObject();
+        String[] requestedPath = new String[]{"mylsp", "nonexistant"};
+
+        JsonElement result = SettingsHelper.findSettings(requestedPath, jsonObject);
+
+        assertNull(result);
+    }
+
+    public void testGetSettingsEmptyJson() {
+        JsonObject jsonObject = JsonParser.parseReader(new StringReader("{}")).getAsJsonObject();
+        String[] requestedPath = new String[]{"mylsp", "myobjectsettings"};
+
+        JsonElement result = SettingsHelper.findSettings(requestedPath, jsonObject);
+
+        assertNull(result);
+    }
+
+    public void testGetSettingsIdentityOnEmptySections() {
+        JsonObject jsonObject = JsonParser.parseReader(new StringReader(testJson)).getAsJsonObject();
+        String[] requestedPath = new String[0];
+
+        JsonElement result = SettingsHelper.findSettings(requestedPath, jsonObject);
+
+        assertEquals(jsonObject, result);
+    }
+}
```

Comment:
Please add findSettings tests with one section and three section (today your test is with zero or two sections)

Reply:
@JanThomas118 is there any chance that you add those tests?

Reply:
It would be nice also to add test with zero and for sections to cover all usecases.

@JanThomas118 if you have no time to do that, please tell me. We could merge your PR and I will add tests.

Reply:
Zero sections is handled by testGetSettingsIdentityOnEmptySections. Currently there are tests for selections with length 0 1 2 and 3. If you want one with four i can add that but what benefit do you hope this brings when 2 and 3 is already covered?

Reply:
I suggested to add  four to cover the case where sections.length > flattered JSON properties lengh.

But it is OK.